packages/network-console-shared/src/file_io/native/collection-adapter.ts

```diff
@@ -0,0 +1,212 @@
+        try {
+            const parsed = JSON.parse(fileContents) as Partial<INCNativeRoot>;
+            if (!('meta' in parsed)) {
+                return {
+                    valid: false,
+                    error: 'No metadata about the file.',
+                };
+            }
+            if (!('networkConsoleCollectionVersion' in parsed.meta!)) {
+                return {
+                    valid: false,
+                    error: 'No version key for this file.',
+                };
+            }
+            if(!('name' in parsed)) {
+                return {
+                    valid: false,
+                    error: 'No name for this file.',
+                };
+            }
+            if (!('entries' in parsed) || !Array.isArray(parsed.entries)) {
+                return {
+                    valid: false,
+                    error: 'Expected "entries" list was missing or not an array.',
+                };
+            }
+
+            return {
+                valid: true,
+                parsed: parsed as INCNativeRoot,
+            };
+        }
+        catch (e) {
+            return {
+                valid: false,
+                error: 'Invalid JSON: ' + e.message,
+            };
+        }
+    }
+
+    get isDirty() {
+        return this._dirty;
+    }
+
+    get name() {
+        return this._current.name;
+    }
+
+    set name(value: string) {
+        this._current.name = value;
+        this._dirty = true;
+    }
+
+    get authorization() {
+        if (!this._current.auth) {
+            this._current.auth = {
+                type: 'inherit',
+            };
+        }
+        return new ContainerAuthorizationAdapter(this._current, () => { this._dirty = true; });
+    }
+
+    set authorization(value: INetConsoleAuthorization) {
+        if (!this._current.auth) {
+            this._current.auth = {
+                type: 'inherit',
+            };
+        }
+        migrateAuthorization(this._current.auth, value);
+    }
+
+    get childEntryIds() {
+        return Array.from(this._keyToIndex.keys());
+    }
+
+    getEntryById(id: string): ICollectionEntryAdapter | null {
+        const index = this._keyToIndex.getByKey(id);
+        if (typeof index !== 'number') {
+            return null;
+        }
+
+        const entry = this._current.entries[index] as NCChild;
+        if ('request' in entry) {
+            return new RequestAdapter(this.format, this, id, entry, () => { this._dirty = true; });
+        }
+        else if ('entries' in entry && Array.isArray(entry.entries)) {
+            return new ContainerAdapter(this.format, this, id, entry, () => { this._dirty = true; });
+        }
+
+        throw new Error(`Parser error: Child with id "${id}" was not a valid request or collection folder.`);
+    }
+
+    async appendContainerEntry(name: string): Promise<ICollectionContainerAdapter> {
+        const folder: INCNativeFolder = {

```

Comment:
While this is true, an incoming change supporting Environments will also reference "containers" because they don't have a hierarchical structure (there's just a single level container).

I'm going to leave this for now. I will continue considering this however.